root/components/relationships-table.tt

```diff
@@ -1,50 +1,82 @@
+                    IF !has.attribute && rel.link.attributes.size; has.attribute = 1; END;
+                    IF !has.artist    && rel.target.artist_credit; has.artist    = 1; END;
+                    IF !has.length    && rel.target.length;        has.length    = 1; END;
+                    LAST IF has.credit && has.attribute && has.artist && has.length;
+                END;
+                LAST IF has.credit && has.attribute && has.artist && has.length;
+            END;
+            has.credit = (has.credit && show_credits) ? 1 : 0;
+            columns_subcount = 1 + has.credit + has.attribute + has.artist + has.length; %]
         <thead>
             <tr>
                 <th>[% l('Date') %]</th>
                 <th>[% l('Title') %]</th>
-                [% IF show_credits -%]<th>[% l('Credited As') %]</th>[%- END %]
+                [%- IF has.credit %]
+                <th>[% l('Credited As') %]</th>
+                [%- END %]
+                [%- IF has.attribute %]
                 <th>[% l('Attributes') %]</th>
+                [%- END %]
+                [%- IF has.artist %]
                 <th>[% l('Artist') %]</th>
+                [%- END %]
+                [%- IF has.length %]
+                <th>[% l('Length') %]</th>
+                [%- END %]
             </tr>
         </thead>
         <tbody>
-        [%- FOREACH relationship IN entity.appearances.pairs -%]
-                <tr class="subh">
-                    <th></th>
-                    <th colspan="[% show_credits ? 4 : 3 %]">[% l(relationship.key) %]</th>
-                </tr>
-                [% FOR rel IN relationship.value %]
-                <tr class="[% loop.parity %]">
-                    <td>[% rel.link.formatted_date %]</td>
-                    <td>
-                        [%~ '<span class="mp mp-rel">' IF rel.edits_pending ~%]
-                        [% link_entity(rel.target, 'show', '', rel.target_credit) %]
-                        [%~ '</span>' IF rel.edits_pending ~%]
-                    </td>
-                    [%~ IF show_credits ~%]
-                    <td>
-                        [%~ rel.source_credit IF rel.source_credit ~%]
-                    </td>
+        [%- FOREACH relationship IN entity.appearances.pairs %]
+            <tr class="subh">
+                <th></th>
+                <th colspan="[% columns_subcount %]">[% l(relationship.key) %]</th>
+            </tr>
+            [%- FOR rel IN relationship.value %]
+            <tr class="[% loop.parity %]">
+                <td>[% rel.link.formatted_date %]</td>
+                <td>
+                    [%~ '<span class="mp mp-rel">' IF rel.edits_pending ~%]
+                    [% link_entity(rel.target, 'show', '', rel.target_credit) %]
+                    [%~ '</span>' IF rel.edits_pending ~%]
+                </td>
+                [%- IF has.credit %]
+                <td>
+                    [%~ rel.source_credit IF rel.source_credit ~%]
```

Comment:
Perhaps this should be fixed to use the same check as above, `rel.source_credit != ''`. Seems like it'll break with "0", as usual.